### PR TITLE
Enerko 1.3 create player

### PR DESCRIPTION
**Pull request by enerko**
- Added a temporary default game frame that runs from main
- Added a controllable player, which looks like a plain white square for now
- Changed GamePanelUI to GamePanel and createGamePanelUI to createGamePanel to avoid confusion
- Wrote a few tests and will be adding more in the future (including testing with a robot class?)

Not too sure if the setter methods should be in the UpdatePlayer class. If you have any opinions/suggestions on how I could improve my code, please don't hesitate to share :)

### CODE REVIEW

File: src/test/java/default_game/PlayerTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package default_game;
+
+import com.sun.tools.javac.Main;
+import entities.default_game.Player;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import use_cases.default_game.UpdatePlayer;
+
```
Comment (FFFiend) [inline src/test/java/default_game/PlayerTest.java:8]: Good test quality.

File: src/main/java/adapters/default_game/GamePanelPresenter.java [hunk 2]
```diff
@@ -1,28 +1,89 @@
...
 
-import use_cases.default_game.IGamePanelPresenterUpdater;
+import use_cases.default_game.UpdatePlayer;
 
 import javax.swing.*;
 import java.awt.*;
 
 /**
- * A class responsible for what the GamePanelUI presents 
+ * A class responsible for what the GamePanelUI presents
```
Comment (jolive3000) [inline src/main/java/adapters/default_game/GamePanelPresenter.java:9]:
```suggestion
 * A class responsible for what the GamePanel presents
```

File: src/main/java/Main.java [hunk 3]
```diff
@@ -30,4 +32,11 @@ private static void tempEditorRunner() {
         CustomGamePresenter tempPresenter = new CustomGamePresenter();
         tempPresenter.callCustomGamePanel("CustomGameEditorPanel");
     }
+    /**
```
Comment (jolive3000) [inline src/main/java/Main.java:35]:
```suggestion

    /**
```

File: src/main/java/user_interface/default_game/GamePanel.java [hunk 4]
```diff
@@ -23,6 +24,12 @@ public void createGamePanelUI() {
...
         window.setVisible(true);
+
+        presenter.startGameThread();
+
+        GamePanelController controller = new GamePanelController();
+        window.addKeyListener(controller);
+        window.setFocusable(true);
     }
 
 }
```
Comment (jolive3000) [inline src/main/java/user_interface/default_game/GamePanel.java:35]: extra space

**Review by jolive3000 — DISMISSED** [review]
It looks great! 

I couldn't find any problems with CA, solid or style except for the comments I made. There are a few missing Javadocs but the methods are self explanatory so it should be fine and you accounded for testing already. 

As for the setPlayer methods, I think the way you did it makes sense.

**Review by tseamuscorlett — DISMISSED** [review]
Looks good!
Glad to see everything is coming together :)

**Comment by enerko** [comment]
Made some changes as requested!

File: src/main/java/user_interface/default_game/GamePanel.java (hunk 4 above)
Comment (FFFiend) [reply src/main/java/user_interface/default_game/GamePanel.java:35]: that startGameThread() method call got me excited I'm not going to lie :D

**Review by FFFiend — APPROVED** [review]

**Pull request merged**